concourse/pipelines/deploy-cloudfoundry.yml

```diff
@@ -0,0 +1,231 @@
+resources:
+  - name: cf-deployment
+    type: bosh-deployment
+    source:
+      target: https://10.0.0.6:25555
+      username: admin
+      password: {{bosh_password}}
+      deployment: {{deploy_env}}
+      ignore_ssl: true
+
+  - name: paas-cf
+    type: git
+    source:
+      uri: https://github.com/alphagov/paas-cf.git
+      branch: {{branch_name}}
+
+  - name: vpc-tfstate
+    type: s3
+    source:
+      bucket: {{state_bucket}}
+      region_name: {{aws_region}}
+      access_key_id: {{aws_access_key_id}}
+      secret_access_key: {{aws_secret_access_key}}
+      versioned_file: vpc.tfstate
+
+  - name: bosh-tfstate
+    type: s3
+    source:
+      bucket: {{state_bucket}}
+      region_name: {{aws_region}}
+      access_key_id: {{aws_access_key_id}}
+      secret_access_key: {{aws_secret_access_key}}
+      versioned_file: bosh.tfstate
+
+  - name: concourse-tfstate
+    type: s3
+    source:
+      bucket: {{state_bucket}}
+      access_key_id: {{aws_access_key_id}}
+      secret_access_key: {{aws_secret_access_key}}
+      versioned_file: concourse.tfstate
+      region_name: eu-west-1
+
+  - name: cf-tfstate
+    type: s3
+    source:
+      bucket: {{state_bucket}}
+      access_key_id: {{aws_access_key_id}}
+      secret_access_key: {{aws_secret_access_key}}
+      versioned_file: cf.tfstate
+      region_name: eu-west-1
+
+  - name: cf-manifest
+    type: s3
+    source:
+      bucket: {{state_bucket}}
+      region_name: {{aws_region}}
+      access_key_id: {{aws_access_key_id}}
+      secret_access_key: {{aws_secret_access_key}}
+      versioned_file: cf-manifest.yml
+
+jobs:
+  - name: s3init
+    serial_groups: [ deploy ]
+    serial: true
+    plan:
+      - get: paas-cf
+      - task: s3init
+        config:
+          image: docker:///governmentpaas/curl-ssl
+          params:
+            AWS_ACCESS_KEY_ID: {{aws_access_key_id}}
+            AWS_SECRET_ACCESS_KEY: {{aws_secret_access_key}}
+          run:
+            path: sh
+            args:
+            - -e
+            - -c
+            - |
+              paas-cf/concourse/scripts/s3init.sh {{state_bucket}} cf.tfstate paas-cf/concourse/init_files/terraform.tfstate.tpl
+          inputs:
+          - name: paas-cf
+      - put: cf-tfstate
+        params:
+          file: s3init/cf.tfstate
```

Comment:
The script has been changed to write to local. The intention of @combor is have a pure pipeline with the dependencies well defined, where the tfstate is passed as a resource between all the jobs.

I think is good, because then we have a 3rd pattern to handle the s3 resource initialisation. We should discuss later which one adopt.